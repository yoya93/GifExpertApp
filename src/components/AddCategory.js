import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [imputValue, setImputValue] = useState("");

  const handleImputChange = (e) => {
    setImputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    if (imputValue.trim().length > 2) {
      setCategories((cate) => {
        setCategories([imputValue, ...cate]);
        setImputValue("");
      });
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{imputValue}</p>
      <input type="text" value={imputValue} onChange={handleImputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
