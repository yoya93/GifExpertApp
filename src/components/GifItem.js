import React from "react";
import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="card  animate__heartBeat">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

// PrimeraApp.defaultProps = {
//   title: "titulo",
//   url: "https://youtube.com",
// };
