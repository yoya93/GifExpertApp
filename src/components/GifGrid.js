import React from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "./hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h2 className="animate__wobble">{category}</h2>

      {loading && <p> Loading</p>}

      <div className="card-grid">
        {images.map((img) => {
          return <GifItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
