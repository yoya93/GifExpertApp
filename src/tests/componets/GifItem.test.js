import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifItem } from "../../components/GifItem";

describe("Pruebas en <GifItem />", () => {
  const url = "https://youtube.com";
  const title = "titulo";
  const wrapper = shallow(<GifItem url={url} title={title} />);

  test("debe de mostrar <GifItem /> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("dede mostrar el title en <p>", () => {
    const p = wrapper.find("p").text().trim();

    expect(p).toBe(title);
  });

  test("debe tener en la imagen el alt y el src ", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe tener el div la clase:  animate__heartBeat", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__heartBeat")).toBe(true);
  });
});
