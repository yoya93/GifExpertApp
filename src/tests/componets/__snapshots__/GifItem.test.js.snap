// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de mostrar <GifItem /> correctamente 1`] = `
<div
  className="card  animate__heartBeat"
>
  <img
    alt="titulo"
    src="https://youtube.com"
  />
  <p>
    titulo
  </p>
</div>
`;
